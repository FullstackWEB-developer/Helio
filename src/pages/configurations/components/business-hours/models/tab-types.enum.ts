export enum TabTypes {
    GeneralSetting = 0,
    SetBusinessHours,
    SetHoliday,
}
