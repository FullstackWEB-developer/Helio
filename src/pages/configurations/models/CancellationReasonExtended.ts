import { CancellationReasonTypes } from "@pages/external-access/models/cancellation-reason-types.enum";

export interface CancellationReasonExtended {
    description?: string;
    existsOnEMR: boolean;
    id: string;
    intentName?: string;
    isDefault: boolean;
    isMapped: boolean;
    name: string;
    nameOnEMR?: string;
    type?: CancellationReasonTypes,
    createdByName: string,
    createdOn: Date,
    modifiedByName: string,
    modifiedOn?: Date
}