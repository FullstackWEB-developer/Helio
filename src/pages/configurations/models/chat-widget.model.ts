import { DisplayPosition } from "./display-position.enum";

export interface ChatWidgetModel
{
    displayPosition: DisplayPosition,
    domains: string[],
    autoStartEnabled: boolean,
    autoStartDelay: number
}