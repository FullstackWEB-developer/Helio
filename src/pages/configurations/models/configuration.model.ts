export interface ConfigurationModel {
    id: string;
    name: string;
    description: string;
    existsOnEMR: boolean;
    isMapped: boolean;
}
