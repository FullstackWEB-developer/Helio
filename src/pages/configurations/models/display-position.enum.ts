export enum DisplayPosition {
    RM = 1,
    RB
}
