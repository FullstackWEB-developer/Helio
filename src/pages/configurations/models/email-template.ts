export interface EmailTemplate {
    id: string;
    subject: string;
    templateBody: string;
    layoutName: string;
    name: string;
    description: string;
    title: string;
    defaultBody: string;
    body: string;
    createdByName: string;
    createdOn: Date;
    modifiedByName?: string;
    modifiedOn?: Date;
}