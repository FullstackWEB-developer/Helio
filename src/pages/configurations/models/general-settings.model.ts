export interface GeneralSettingsModel {
    forceToRedirect: boolean;
    redirectToExternalPhone: string;
    infoEmailAddress: string;
    infoPhoneNumber: string;
    deletedDepartments: string;
}