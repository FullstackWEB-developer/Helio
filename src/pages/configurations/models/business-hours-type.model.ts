import { DayOfWeek } from '@shared/models/DayOfWeek';

export enum WorkingHoursType {
  _24_7 = 1,
  Custom = 2,
}

export interface OperationSettingModel {
  id: number;
  timeZone: string;
  voiceRouting: boolean;
  autoReplySms: boolean;
  autoReplyEmail: boolean;
  routingNumber: string;
  workingHoursType: WorkingHoursType;
  workingHours: WorkingHourModel[];
  offDates: WorkingOffDateModel[];
}

export interface WorkingHourModel {
  day: DayOfWeek;
  startTime: string;
  endTime: string;
}

export interface WorkingOffDateModel {
  description: string;
  startDateTime: string;
  endDateTime: string;
}
