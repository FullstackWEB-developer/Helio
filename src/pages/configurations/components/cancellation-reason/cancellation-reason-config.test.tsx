import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import CancellationReasonConfig from './cancellation-reason-config';
import { act } from 'react-dom/test-utils';
import Api from '@shared/services/api';
import { CancellationReasonExtended } from '@pages/configurations/models/CancellationReasonExtended';
import { CancellationReasonTypes } from '@pages/external-access/models/cancellation-reason-types.enum';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("CancellationReasonConfig tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders CancellationReasonConfig correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CancellationReasonConfig/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders CancellationReasonConfig with data correctly", async () => {
        var response = {
            data: [{
                description: "string",
                existsOnEMR: true,
                id: 1,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 2,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 3,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 4,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 5,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 6,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 7,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 8,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            },{
                description: "string",
                existsOnEMR: true,
                id: 9,
                intentName: "string",
                isDefault: false,
                isMapped: false,
                name: "string",
                nameOnEMR: "string",
                type: CancellationReasonTypes.Cancel,
                createdByName: "string",
                createdOn: new Date(),
                modifiedByName: "string",
                modifiedOn: new Date()
            }] as CancellationReasonExtended[] 
        } as any
        jest.spyOn(Api, 'get').mockResolvedValue(response.data);
        await act(async () => {
            render(<TestWrapper mockState={mockState}>
                <CancellationReasonConfig/>
            </TestWrapper>);
        });
        expect(Api.get).toHaveBeenCalledTimes(2);
    });
})