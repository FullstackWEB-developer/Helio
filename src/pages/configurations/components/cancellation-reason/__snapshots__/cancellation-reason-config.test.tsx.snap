// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancellationReasonConfig tests renders CancellationReasonConfig correctly 1`] = `
<DocumentFragment>
  <div
    class="px-6 pt-7 flex flex-1 flex-col group overflow-y-auto"
  >
    <h6
      class="pb-7"
    >
      configuration.cancellation_reason.title
    </h6>
    <div
      class="body2 whitespace-pre-line pb-14"
    >
      configuration.cancellation_reason.description
    </div>
    <div
      class="h-12 px-4 cancellation-reason-list-grid head-row caption-caps"
    >
      <div
        class="truncate relative w-full h-full inline-block"
      >
        <div
          class=""
        >
          configuration.cancellation_reason.grid_id
        </div>
      </div>
      <div
        class="truncate relative w-full h-full inline-block"
      >
        <div
          class=""
        >
          configuration.cancellation_reason.grid_name
        </div>
      </div>
      <div
        class="truncate relative w-full h-full inline-block"
      >
        <div
          class=""
        >
          configuration.cancellation_reason.grid_description
        </div>
      </div>
      <div />
    </div>
    <div
      class="flex flex-col space-y-4 items-center justify-center px-2"
    >
      <div>
        <div
          class=""
        >
          <svg
            class="spinner icon-medium "
            viewBox="0 0 40 40"
          >
            <circle
              class="path"
              cx="20"
              cy="20"
              fill="none"
              r="16"
              stroke-width="4"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
